compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -3752,6 +3753,19 @@ public void visit(BLangTableTypeNode tableType) {
     }
 
     public void visit(BLangErrorType errorType) {
+        BLangType detailType = errorType.detailType;
+        if (detailType != null && detailType.getKind() == NodeKind.CONSTRAINED_TYPE) {
+            BLangType constraint = ((BLangConstrainedType) detailType).constraint;
+            if (constraint.getKind() == NodeKind.USER_DEFINED_TYPE) {
+                BLangUserDefinedType userDefinedType = (BLangUserDefinedType) constraint;
+                if (userDefinedType.typeName.value.equals(TypeDefBuilderHelper.INTERSECTED_ERROR_DETAIL)) {
+                    // skip this as this is special case added to support error intersection where detail
```

Comment:
This is because here [https://github.com/ballerina-platform/ballerina-lang/pull/31731/files#diff-c962be7fdc32ce89c2c022fcbf1288f82f98f9ee1ee2d7323f75dbc45b3b9363R244] I have special cased the type-def to avoid large desugaring. Then I catch that here https://github.com/ballerina-platform/ballerina-lang/pull/31731/files#diff-fa94fe90a875c08eaf806f89f6ceb7eec7205acaf75c88a41e3f7103abfa5a52R3614 and set the detail type node to `null`

One of the problem I'm trying to avoid by this is, say the constraint of an error detail map is `lang.value:Cloneable` and we don't have an import for that module, if I didn't do this when this new type get resolved, the resolver can't find the module `lang.value`.

Reply:
As this is just a reference to some other type (`type A foo:Bar`) I don't think we will miss anything because of skipping this.